src/api/java/io/github/cvc5/Proof.java

```diff
@@ -20,39 +20,28 @@
 /**
  * A cvc5 Proof.
  */
-public class Proof implements IPointer
+public class Proof extends AbstractPointer implements Comparable<Proof>
```

Comment:
IPointer is necessary for avoiding memory leak and Comparable<Proof> is not needed here. To pass coverage tests, the method equals should be explicitly called.

Reply:
AbstractPointer implements IPointer. This is also what we do for sorts and terms. Operator `==` should call `equals`, else we get an issue in the future if we change the comparison from pointer comparison to something more sophisticated (not that I'm saying that we plan to do this in the near future).

Reply:
Resolved this offline.